code/modules/mob/dead/observer/observer.dm

```diff
@@ -89,6 +89,7 @@ GLOBAL_VAR_INIT(observer_default_invisibility, INVISIBILITY_OBSERVER)
 
 	//starts ghosts off with all HUDs.
 	toggle_all_huds_on(body)
+	addtimer(CALLBACK(src, .proc/set_ghost_darkness_level))
```

Comment:
Invoke async please

Reply:
`set_ghost_darkness_level` has to run after this one ends, not concurrently. Since the ghost has no client at this point.

Reply:
Put this below 93 (the `..()`)

Reply:
That does not work either, the client is not put in the observer mob, even after the `..()`

Reply:
The `body` arg should have the client attached. That's what I had to do with the `toggle_all_huds_on(body)` proc in  #16459.

Reply:
It does, and that's good. But unfortunately the mob doesn't have a `hud_used`, and we need that to call `update_sight` because it calls `sync_lighting_plane_alpha`, that requires `hud_used`